text/0063-attestation-grpc-apis.md

```diff
@@ -0,0 +1,316 @@
+attestation protocol contains different attestation evidence than the previous
+[EPID](https://api.trustedservices.intel.com/documents/sgx-attestation-api-spec.pdf)
+protocol. The use of an opaque buffer of bytes prevents the recipient of the
+attestation evidence from knowing if the evidence is for EPID or DCAP.
+
+We want to provide an alternative gRPC API that will be expandable to better
+support future attestation evidence format updates.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+An _enclave_ is the name given for an application running in an Intel SGX
+[trusted execution environment](https://en.wikipedia.org/wiki/Trusted_execution_environment).
+Enclave's provide attestation evidence to requesting clients. This attestation
+evidence provides assurance that the enclave has not been modified and that the
+hardware the enclave is running on is not subject to any unnacceptable
+configurations (hyperthreading is disabled, on board GPU is disabled, etc).
+
+Prior to trusted communication with an encalve, a client will request the
+attestation evidence and verify it's correctness. Once verified, the client will
+use an encryption key provided in the attestation evidence to further
+communicate with the enclave in a trusted manner.
+
+## Creating Attestation Evidence
+
+The enclave to be verified will communicate with a trusted quoting enclave (QE)
+in the same machine. This quoting enclave will sign the evidence provided by the
+enclave to be verified. This signed evidenced is refered to as a _quote_.  The
+quote is sent to an attestation service which will verify the quoting enclave's
+signature. This service will verify and sign the quote, creating the attesation
+evidence that clients expect.
+
+```mermaid
+sequenceDiagram
+    Client->>+Enclave: Request attestation evidence
+    box Running Server
+    Participant Enclave
+    Participant Quoting Enclave
+    end
+    autonumber
+    Enclave->>+Quoting Enclave: Enclave evidence
+    Quoting Enclave->>+Enclave: Quote
+    Enclave->>+Attestation Service: Quote
+    Attestation Service->>+Enclave: Attestation evidence
+    Enclave->>+Client: Attestation evidence
+    Client->>Client: Verify attestation evidence
+    loop Trusted Communication
+        Client-->>Enclave: Secure message
+        Enclave-->>Client: Secure message
+    end
+```
+
+The purpose of this MCIP is to better define the data format of the attestation
+evidence that enclaves provide to clients.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+There are two common ways of attesting with an enclave:
+
+1. [Client to enclave](#1-client-to-enclave)
+2. [Enclave to enclave](#2-enclave-to-enclave)
+
+The provided services support these two ways:
+
+```prost
+service AttestService {
+    rpc Attest(AttestRequest) returns (AttestResponse);
+    rpc MutualAttest(MutualAttestRequest) returns (MutualAttestResponse);
```

Comment:
Shortened the types in here, from what's defined at https://github.com/mobilecoinfoundation/protobufs/blob/main/mobilecoinfoundation/attestation/v1/services.proto#L8
Not sure if the sure if we want to keep the longer types.

Reply:
Created #64